lua/terrortown/entities/roles/spy/shared.lua

```diff
@@ -70,7 +70,7 @@ if CLIENT then
 	end)
 else
 	hook.Add("TTT2SpecialRoleSyncing", "TTT2RoleSpyMod", function(ply, tbl)
-		if ply and not ply:HasTeam(TEAM_TRAITOR) or GetRoundState() == ROUND_POST then return end
+		if ply and not ply:HasTeam(TEAM_TRAITOR) or ply:GetSubRole() == ROLE_BODYGUARD or GetRoundState() == ROUND_POST then return end

```

Comment:
It's quite late, but I would propose a more generic version: Please use the unknownTeam attribute to check if the spy should share its role information instead of checking for the bodyguard role.

Reply:
Spy already has unkownTeam to true?

Reply:
No, you should check in this line whether ply:GetSubRoleData().unknownTeam is true instead of checking for the role Bodyguard explicitly

Reply:
Oh thats what you mean. Sure yeah.